### ISSUE TITLE
Piece detection not working

### ISSUE THREAD

**Issue by leason** [aelmiger/chessboard2fen#9]
Hi, not sure if this is still a side project or not.  I'm trying to build a little bot that will let me play an over-the-board game with chess.com bots or my friends using raspberry pi parts.

I got your code working on my macbook but the detection doesn't seem to be working for the pieces.  It looks like it draws the board corners and box right, but the pieces don't get detected at all.  Anyone else run into that?  No errors, just a warning about the tensor model but doesn't seem like that would cause the problem: WARNING:tensorflow:No training configuration found in save file, so the model was *not* compiled. Compile it manually.

**Comment by aelmiger** [comment]
What images are you testing with?

**Comment by leason** [comment]
I've tested both with my own and with the included test images.

Lee

**Comment by aelmiger** [comment]
Do you have any more information regarding what is not working exactly?

**Comment by leason** [comment]
I'm not really sure how to troubleshoot this issue.  The bounding box
appears on the board outline from the corner detection step, so I'm
confident the packages are installed right.  But none of the test images
show bounding boxes on the pieces, and the generated chess board image is
always the starting position.  I don't know if it's a problem with the
chess piece model or something like that?  I confirmed the filename is
right and there's no error messages in the output terminal.

Any pointers on things to try?

**Comment by leason** [comment]
After a lot of experimentation with this and several other projects what I'm learning is that these models will only work reliably with the exact chess set they were trained on.  I'd be willing to just buy the set you used for training if you can point me at the online store you got it from?  My focus is more on building a companion bot to play an over-the-board game against so I'm trying to avoid having to take the time right now to train my own model if I can avoid it.

**Comment by aelmiger** [comment]
Yes, that is correct. The CNN was only trained with one specific board and will most likely fail on other boards without additional training data.

Unfortunately I cannot find out which chessboard I used for training.

**Comment by leason** [comment]
Ok, thank you for the time and responses in any case.  I don't know
anything about how to train these kinds of models so I'll go start doing
some research.  Cheers!
